### PR TITLE
Use hash_map::Entry to save a map lookup

### PR DESCRIPTION
**Pull request by tamird**
A bit of golf noticed via https://medium.com/programming-servo/programming-servo-workers-at-your-service-db71e5943511.

r? @gterzian

### CODE REVIEW

File: components/constellation/constellation.rs [hunk 1]
```diff
@@ -2623,32 +2623,30 @@ where
...
 
     fn handle_register_serviceworker(&mut self, scope_things: ScopeThings, scope: ServoUrl) {
-        let origin = scope.origin();
-
-        if let Some(mgr) = self.sw_managers.get(&origin) {
-            let _ = mgr.send(ServiceWorkerMsg::RegisterServiceWorker(scope_things, scope));
-        } else {
-            let (own_sender, receiver) = ipc::channel().expect("Failed to create IPC channel!");
+        // This match is equivalent to Entry.or_insert_with but allows for early return.
+        let _ = match self.sw_managers.entry(scope.origin()) {
```
Comment (gterzian) [inline components/constellation/constellation.rs:2627]: Ok, how about assigning to something like `sw_sender`, and then using that to send the message?

**Review by gterzian — CHANGES_REQUESTED** [review]
Nice catch! 

Could we perhaps still assign the result of the match and call `send` on it? I think it would be clearer than the currently proposed `let _ = match {}.send`.

To fix the formatting, you can also do `./mach fmt`.

**Comment by tamird** [comment]
Thanks, done!

**Comment by tamird** [comment]
At last, this is green.

**Comment by gterzian** [comment]
Thanks!

@bors-servo r+

**Pull request merged**